pkg/controller/v1alpha1/inferencegraph/raw_ig.go

```diff
@@ -98,6 +104,20 @@ func createInferenceGraphPodSpec(graph *v1alpha1.InferenceGraph, config *RouterC
 		}
 	}
 
+	// If auth is enabled for the InferenceGraph:
+	// * Add --enable-auth argument, to properly secure kserve-router
+	// * Add the --inferencegraph-name argument, so that the router is aware of its name
+	// * Enable auto-mount of the ServiceAccount, because it is required for validating tokens
+	// * Set a non-default ServiceAccount with enough privileges to verify auth
+	if graph.HasAuthEnabled() {
+		podSpec.Containers[0].Args = append(podSpec.Containers[0].Args, "--enable-auth")
+
+		podSpec.Containers[0].Args = append(podSpec.Containers[0].Args, "--inferencegraph-name")
+		podSpec.Containers[0].Args = append(podSpec.Containers[0].Args, graph.GetName())
+
+		podSpec.AutomountServiceAccountToken = proto.Bool(true)
```

Comment:
I think I've already asked you about it, but is the auto mount required for IG?
If so, would you mind adding a comment here explaining why?

Reply:
When auth is enabled, yes. Because the kserve-router will be doing calls to the cluster API. I'll add the comment.

Reply:
Sounds good, thanks.